lib/opencensus/trace.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Opencensus.Trace do
+  @doc "Wraps the given block in a tracing child span with the given label/name"
+  defmacro with_child_span(label, do: block) do
+    quote do
+      :ocp.with_child_span(unquote(label), %{})
+
+      try do
+        unquote(block)
+      after
+        :ocp.finish_span()
```

Comment:
Use `:oc_trace.finish_span/1` to be sure that the span you opened on L5 is the same that you close there.

Reply:
Ah yea, I'll fix this up.